bridgerelayer/chains/ethereum/router.go

```diff
@@ -40,37 +46,36 @@ func (er Router) Route(eventData types.EventData) error {
 }
 
 // BuildPacket builds a data packet from tx data
-func (er Router) buildPacket(id common.Address, eLog ctypes.Log) (types.Packet, error) {
-	appAddress := id.Bytes()
-	var appID [32]byte
-	copy(appID[:], appAddress)
-
+func (er Router) buildPacket(id common.Address, eLog ctypes.Log) (types.PacketV2, error) {
 	// RLP encode event log's Address, Topics, and Data
 	var buff bytes.Buffer
 	err := eLog.EncodeRLP(&buff)
 	if err != nil {
-		return types.Packet{}, err
+		return types.PacketV2{}, err
 	}
 
 	// Generate a proof by signing a hash of the encoded data
 	proof, err := prover.GenerateProof(buff.Bytes(), er.keybase.PrivateKey())
 	if err != nil {
-		return types.Packet{}, err
+		return types.PacketV2{}, err
 	}
 
-	// Construct and wrap message
-	message := types.NewMessage(buff.Bytes(), proof)
-	wrappedMessage := types.NewLightClientProof(message)
-
-	packet := types.NewPacket(appID, wrappedMessage)
+	packet := types.PacketV2{
+		Data: buff.Bytes(),
+		Signature: proof.Signature,
+	}
 	return packet, nil
 }
 
 // SendPacket sends a tx data packet to the bridge
-func (er Router) sendPacket(packet types.Packet) error {
+func (er Router) sendPacket(appID [32]byte, packet types.PacketV2) error {
 	log.Info("Sending packet:\n", packet)
 
-	// Bridge.Send(packet.AppID, packet.Message)
+	client, err := substrate.NewClient()
+	if err != nil {
+		panic(err)
```

Comment:
I think it makes more sense to route messages through a single Substrate client that's initialized once when the Relayer is started instead of setting up a new client for each request. If the Substrate client is initialized once then it makes sense to use panic, but here we should be returning an error.

Returning the error is sufficient for now, we can refactor the Substrate client later.

Reply:
fixed in latest upload.